numba/cuda/cudaimpl.py

```diff
@@ -100,29 +100,13 @@ def ptx_sreg_impl(context, builder, sig, args):
 def ptx_cmem_arylike(context, builder, sig, args):
     lmod = builder.module
     [arr] = args
-    flat = arr.flatten(order='A')
     aryty = sig.return_type
-    dtype = aryty.dtype
-
-    if isinstance(dtype, types.Complex):
-        elemtype = (types.float32
-                    if dtype == types.complex64
-                    else types.float64)
-        constvals = []
-        for i in range(flat.size):
-            elem = flat[i]
-            real = context.get_constant(elemtype, elem.real)
-            imag = context.get_constant(elemtype, elem.imag)
-            constvals.extend([real, imag])
-
-    elif dtype in types.number_domain:
-        constvals = [context.get_constant(dtype, flat[i])
-                     for i in range(flat.size)]
-
-    else:
-        raise TypeError("unsupport type: %s" % dtype)
 
-    constary = lc.Constant.array(constvals[0].type, constvals)
+    constvals = [
+        context.get_constant(types.byte, i)
+        for i in arr.flatten(order='A').data.tobytes()
```

Comment:
This is causing error on windows:

```
test_const_array (numba.cuda.tests.cudapy.test_constmem.TestCudaConstantMemory)
test_const_array_2d (numba.cuda.tests.cudapy.test_constmem.TestCudaConstantMemory)
test_const_array_3d (numba.cuda.tests.cudapy.test_constmem.TestCudaConstantMemory)
test_const_record (numba.cuda.tests.cudapy.test_constmem.TestCudaConstantMemory)
test_const_record_align (numba.cuda.tests.cudapy.test_constmem.TestCudaConstantMemory)



File "..\_test_env\lib\site-packages\numba\cuda\tests\cudapy\test_constmem.py", line 60:
def cuconstRecAlign(A, B, C, D, E):
    Z = cuda.const.array_like(CONST_RECORD_ALIGN)
    ^
[1] During: lowering "$0.5 = call ptx.cmem.arylike($0.4, kws=[], args=[Var($0.4, c:\conda64\conda-bld\numba_1533830313780\_test_env\lib\site-packages\numba\cuda\tests\cudapy\test_constmem.py (60))], func=ptx.cmem.arylike, vararg=None)" at c:\conda64\conda-bld\numba_1533830313780\_test_env\lib\site-packages\numba\cuda\tests\cudapy\test_constmem.py (60)

'buffer' object has no attribute 'tobytes'
```

Reply:
likely a python2.7 error

Reply:
I've pushed a fix for this - it uses [this](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.ndarray.tobytes.html) instead